Automattic/wp-calypso
# Refactor: Create a new AppPromo component

This PR created a new AppPromo component that is going to be used in a few different places to make it easier to showcase the Jetpack app though calypso. 

<img width="400" alt="Screenshot 2023-10-13 at 3 11 53 PM" src="https://github.com/Automattic/wp-calypso/assets/115071/2e59a31b-1594-45ba-995f-ab560618123c">

Devdocs

<img width="1695" alt="Screenshot 2023-10-13 at 3 13 31 PM" src="https://github.com/Automattic/wp-calypso/assets/115071/9c5e0fdf-f2ea-4072-b839-c4a2bdc09366">


## Proposed Changes

* Refactors the current Go Mobile component that was on the custom home to AppPromo. 
* Adds a new AppPromo component to the dev docs.


## Testing Instructions
1. Apply this Patch. 
2. Visit /home/sitename.com and notice that the Mobile Promo is still the same. 
3. Visit /devdocs/design/app-promo and notice the new devdocs page and how that it works as expected.


## Pre-merge Checklist
